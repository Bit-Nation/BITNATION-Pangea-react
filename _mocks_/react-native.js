module.exports = require('react-native-mock');