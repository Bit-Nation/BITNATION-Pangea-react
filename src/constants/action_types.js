export const CORE_ACTIONS = {
  LOADING: 'LOADING',
  LOADED: 'LOADED',
};

export const UI_ACTIONS = {
  OPEN_DRAWER: 'OPEN_DRAWER',
  CLOSE_DRAWER: 'CLOSE_DRAWER',
  CHANGE_SCREEN: 'CHANGE_SCREEN',
  CHANGE_TITLE: 'CHANGE_TITLE',
};

export const NATIONS_ACTIONS = {
  SET_CURRENT_NATION: 'SET_CURRENT_NATION',
};
