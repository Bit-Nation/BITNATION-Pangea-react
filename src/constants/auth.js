export const LOGIN = 'AUTH/LOGIN';
export const LOGOUT = 'AUTH/LOGOUT';
