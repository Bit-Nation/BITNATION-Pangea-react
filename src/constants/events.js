export const AUTH_DIALOG_OPEN = 'AUTH/DIALOG_OPEN';
export const AUTH_DIALOG_CLOSE = 'AUTH/DIALOG_CLOSE';
