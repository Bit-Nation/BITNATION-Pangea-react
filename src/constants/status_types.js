export const APP_STATUS = {
  INACTIVE: 'INACTIVE',
  RUNNING: 'RUNNING',
  STARTING: 'STARTING',
  BROKEN: 'BROKEN',
};

export const SCREEN_TYPES = {
  WELCOME: 'WELCOME',
  MAIN: 'MAIN',
  NATIONS: 'NATIONS',
  NATION: 'NATION',
};
