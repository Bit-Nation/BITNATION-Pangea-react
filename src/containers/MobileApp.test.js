'use strict';

import React from 'react';
import { shallow } from 'enzyme';
import { APP_STATUS } from '../constants/status_types';
import { Provider } from 'react-redux'; // eslint-disable-line-line import/first
import { Text } from 'react-native';
import configureMockStore from 'redux-mock-store'; // eslint-disable-line-line import/first
import thunk from 'redux-thunk'; // eslint-disable-line-line import/first

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const MobileApp = require.requireActual('./MobileApp.js').default;

describe('<MobileApp />', () => {
  it('should contain the status text', () => {
    const store = mockStore({ 
      core: {
        appStatus: APP_STATUS.INACTIVE,
      },
    });

    const mobileApp = shallow(<Provider store={store}><MobileApp /></Provider>);
    mobileApp.debug();
    expect(mobileApp).toMatchSnapshot();
  });
});
