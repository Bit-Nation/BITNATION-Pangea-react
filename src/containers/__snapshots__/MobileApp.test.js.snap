// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileApp /> should contain the status text 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(App) />,
  "nodes": Array [
    <Connect(App) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Provider
      store={
            Object {
                  "clearActions": [Function],
                  "dispatch": [Function],
                  "getActions": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                }
      }
>
      <Connect(App) />
</Provider>,
    "_instance": Provider {
      "context": Object {},
      "props": Object {
        "children": <Connect(App) />,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "refs": Object {},
      "state": Object {},
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <Connect(App) />,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
>
    <Connect(App) />
</Provider>,
}
`;
